File: Sub Array With Given Sum/SubArray.py
def find_subarray_with_given_sum(array, given_sum):
    start = 0  # Initialize start pointer
    end = 0    # Initialize end pointer
    current_sum = array[0]  # Initialize currentSum with first element

    while end < len(array):
        if current_sum == given_sum:  # If currentSum equals givenSum, subarray found
            print("Subarray found from index", start, "to", end)
            return
        elif current_sum < given_sum:  # If currentSum is less than givenSum, move end pointer forward
            end += 1
            if end < len(array):  # Avoid accessing out of bounds
                current_sum += array[end]
        else:  # If currentSum is greater than givenSum, move start pointer forward
            current_sum -= array[start]
            start += 1

    print("Subarray not found")  # If no subarray found


array = [1, 4, 20, 3, 10, 5]
given_sum = 33

find_subarray_with_given_sum(array, given_sum)

File: Sub Array With Given Sum/SubArray.c
#include <stdio.h>

void findSubarrayWithGivenSum(int array[], int size, int givenSum) {
    int start = 0; // Initialize start pointer
    int end = 0;   // Initialize end pointer
    int currentSum = array[0]; // Initialize currentSum with first element

    while (end < size) {
        if (currentSum == givenSum) { // If currentSum equals givenSum, subarray found
            printf("Subarray found from index %d to %d\n", start, end);
            return;
        } else if (currentSum < givenSum) { // If currentSum is less than givenSum, move end pointer forward
            end++;
            if (end < size) // Avoid accessing out of bounds
                currentSum += array[end];
        } else { // If currentSum is greater than givenSum, move start pointer forward
            currentSum -= array[start];
            start++;
        }
    }
    printf("Subarray not found\n"); // If no subarray found
}

int main() {
    int array[] = {1, 4, 20, 3, 10, 5};
    int size = sizeof(array) / sizeof(array[0]);
    int givenSum = 33;

    findSubarrayWithGivenSum(array, size, givenSum);

    return 0;
}
